st_weaviate_connection/connection.py

```diff
@@ -1,63 +1,204 @@
+        """
+        Initialize the Weaviate connection.
+
+        Parameters
+        ----------
+        connection_name : str
+            The name of the connection.
+        url : str, optional
+            The URL of a Weaviate Cloud cluster.
+            Provide this or the `weaviate_client` parameter.
+            Default: None.
+        api_key : str, optional
+            The Weaviate API key to use for authentication. Default: None.
+        additional_headers : dict, optional
+            Additional headers to include in the request.
+            e.g.: "X-<PROVIDER>-Api-Key": "<API_KEY>".
+            Default: None.
+        """
+
         self.url = url
         self.api_key = api_key
         self.additional_headers = additional_headers
+        if url == "localhost":
+            self._client = weaviate.connect_to_local(
+                auth_credentials=self._create_auth_config(),
+                headers=self.additional_headers,
+                skip_init_checks=True,
+            )
+        else:
+            self._client = weaviate.connect_to_weaviate_cloud(
+                cluster_url=url,
+                auth_credentials=self._create_auth_config(),
+                headers=self.additional_headers,
+                skip_init_checks=True,
+            )
         super().__init__(connection_name, **kwargs)
 
-    def _connect(self, **kwargs) -> Client:
-        auth_config = self._create_auth_config()
-        url = self.url or self._secrets.get("WEAVIATE_URL")
-        return Client(
-            url,
-            auth_client_secret=auth_config,
-            additional_headers=self.additional_headers,
-        )
+    def _connect(self) -> WeaviateClient:
+        self._client.connect()
+        return self._client
 
-    def _create_auth_config(self) -> Optional[weaviate.AuthApiKey]:
+    def _create_auth_config(self) -> Optional[_APIKey]:
         api_key = self.api_key or self._secrets.get("WEAVIATE_API_KEY")
         if api_key is not None:
-            return weaviate.AuthApiKey(api_key=api_key)
+            return Auth.api_key(api_key=api_key)
         else:
             return None
 
-    def _convert_to_dataframe(self, results) -> pd.DataFrame:
-        class_name = list(results["data"]["Get"].keys())[0]
-        data = results["data"]["Get"][class_name]
-        df = pd.json_normalize(data)
+    def _gql_to_dataframe(self, results: _RawGQLReturn) -> pd.DataFrame:
+        collection_name = list(results.get.keys())[0]
+        data = results.get[collection_name]
+        df = pd.DataFrame(data)
         return df
 
-    def query(self, query: str, ttl: int = 3600, **kwargs) -> pd.DataFrame:
-        @cache_data(ttl=ttl)
-        def _query(query: str, **kwargs):
-            results = self._connect().query.raw(query)
-            if "errors" in results:
-                error_message = (
-                    f"The GraphQL query returned an error: {results['errors']}"
-                )
+    def hybrid_search(
```

Comment:
The other db-related streamlit connectors use `query` as the main way to query the database they're connected to. I think we should follow this convention so streamlit users have consistent experience across connectors.

Reply:
Ah I see. Okay, I will update `hybrid_search` to `query` and leave the GQL one as is then.